docs/samples/topic/two-way.md

```diff
@@ -396,4 +396,237 @@ pack.setUserAuthentication({
+      type: coda.ValueType.String,
+      codaType: coda.ValueHintType.SelectList,
+      fromKey: "currency_code",
+      mutable: true,
+      // Dynamic list of options.
+      options: async function (context) {
+        // Fetch the list of supported currencies.
+        let response = await context.fetcher.fetch({
+          method: "GET",
+          url: "https://secure.splitwise.com/api/v3.0/get_currencies",
+        });
+        return response.body.currencies.map(currency => currency.currency_code);
+      },
+    },
+    category: {
+      ...CategorySchema,
+      codaType: coda.ValueHintType.SelectList,
+      mutable: true,
+      // Dynamic list of options, as objects.
+      options: async function (context) {
+        let response = await context.fetcher.fetch({
+          method: "GET",
+          url: "https://secure.splitwise.com/api/v3.0/get_categories",
+        });
+        let categories = response.body.categories;
+        let result = [];
+        for (let category of categories) {
+          result.push(category);
+          let subcategories = category.subcategories || [];
+          for (let subcategory of subcategories) {
+            if (subcategory.name === "Other") {
+              // Add the parent category's name in front, to distinguish it.
+              subcategory.name = `${category.name} - ${subcategory.name}`;
+            }
+            result.push(subcategory);
+          }
+        }
+        return result;
+      },
+    },
+  },
+  displayProperty: "description",
+  idProperty: "expenseId",
+  featuredProperties: [
+    "date", "cost", "notes", "currency", "repeat", "category",
+  ],
+});
+
+pack.addSyncTable({
+  name: "Expenses",
+  description: "Lists your expenses.",
+  identityName: "Expense",
+  schema: ExpenseSchema,
+  formula: {
+    name: "SyncExpenses",
+    description: "Syncs the data.",
+    parameters: [],
+    execute: async function (args, context) {
+      let [] = args;
+      let offset = context.sync.continuation?.offset as number || 0;
+      let limit = 20;
+      let url = coda.withQueryParams(
+        "https://secure.splitwise.com/api/v3.0/get_expenses",
+        {
+          offset: offset,
+          limit: limit,
+        }
+      );
+      let response = await context.fetcher.fetch({
+        method: "GET",
+        url: url,
+      });
+      let expenses = response.body.expenses;
+
+      let continuation;
+      if (expenses.length === limit) {
+        offset += limit;
+        continuation = { offset: offset };
+      }
+
+      return {
+        result: expenses,
+        continuation: continuation,
+      };
+    },
+    maxUpdateBatchSize: 10,
+    executeUpdate: async function (args, updates, context) {
+      // Make all of the updates in parallel to improve performance.
+      let jobs = updates.map(async update => {
+        return updateExpense(context, update);
+      });
+      let completed = await Promise.allSettled(jobs);
```

Comment:
Side note: I don't know if we formed any conclusion on whether whether/how to make a helper function or option to do the `Promise.allSettled` but unless someone has a better way I'd be in favor of making a standard helper function for this at least

Reply:
No objections here!